### PR TITLE
Otb academy - bookshelf

### PR DESCRIPTION
**Pull request by jdks**
A first attempt at the bookshelf Sinatra app.

@jdks
@technancy

### CODE REVIEW

File: features/support/ui_driver.rb [hunk 1]
```diff
@@ -20,5 +20,40 @@ def add_book(title)
...
+    fill_in 'query', with: title
+    click_button 'Search'
+  end
+
+  def add_book(title)
+    library.add_book(title) 
+  end
+
+  def search_results
+    all('.result').map { |element| element.text }
```
Comment (robgough) [inline features/support/ui_driver.rb:45]: This line can be shortened to something like:

`all('.result').map(&:text)`

File: lib/library.rb [hunk 2]
```diff
@@ -14,3 +14,35 @@ def search_by_title(title)
...
 
 end
+
+require 'sinatra'
+configure do
+  set :library, Library.new
+end
+
+get '/' do 
+  %{
```
Comment (robgough) [inline lib/library.rb:24]: It might be worth splitting this out into it's own view.

File: lib/library.rb [hunk 3]
```diff
@@ -14,3 +14,35 @@ def search_by_title(title)
...
+
+  </form>
+</body>
+</html>
+}
+end
+
+post '/' do
+  books = settings.library.search_by_title(params[:booktitle])
+  return_str = "<ul>"
```
Comment (robgough) [inline lib/library.rb:41]: Rather than building up a return string, it would be neater if you created a view for this.

Comment (jdks) [inline lib/library.rb:41]: Thanks Rob.  Made those changes.

**Pull request closed without merging**